vllm_spyre/platform.py

```diff
@@ -73,12 +74,41 @@ def check_and_update_config(cls, vllm_config: VllmConfig) -> None:
             scheduler_config.scheduler_cls = \
                 "vllm_spyre.core.scheduler.SpyreScheduler"
 
-        cache_config = vllm_config.cache_config
-        if cache_config:
-            # spyre needs block_size = max_model_len
-            vllm_config.cache_config.block_size = \
-                vllm_config.model_config.max_model_len
+        # Override --max-num-seqs to the biggest warmup batch size
+        # And override --max-model-len to the biggest warmup sequence
         cls.set_warmup_shapes(scheduler_config)
+        max_batch_size = 0
+        max_seq_len = 0
+        for shape in cls.get_warmup_shapes():
+            max_batch_size = max(max_batch_size, shape['batch_size'])
+            max_seq_len = max(max_batch_size,
+                              shape['prompt_length'] + shape['new_tokens'])
+
+        if envs.VLLM_USE_V1:
+            # The v0 scheduler will run out of blocks if this is overridden
+            scheduler_config.max_num_seqs = max_batch_size
```

Comment:
Why will it run out of blocks? I guess it's not that important.

Reply:
Yeah... I didn't debug too far on that one because it's v0, and I assume our plan is still to focus development only on v1